vm/src/builtins/classmethod.rs

```diff
@@ -53,22 +53,32 @@ impl GetDescriptor for PyClassMethod {
         cls: Option<PyObjectRef>,
         vm: &VirtualMachine,
     ) -> PyResult {
-        let (zelf, obj) = Self::_unwrap(zelf, obj, vm)?;
-        let cls = cls.unwrap_or_else(|| obj.class().clone().into());
-        let callable = zelf.callable.lock().clone();
-        Ok(PyBoundMethod::new_ref(cls, callable, &vm.ctx).into())
+        let (zelf, _obj) = Self::_unwrap(zelf, obj, vm)?;
+        let cls = cls.unwrap_or_else(|| _obj.class().clone().into());
+        let _descr_get: PyResult<PyObjectRef> = zelf.callable.lock().get_attr("__get__", vm);
+        match _descr_get {
+            Err(_) => Ok(PyBoundMethod::new_ref(cls, zelf.callable.lock().clone(), &vm.ctx).into()),
+            Ok(_descr_get) => vm.invoke(&_descr_get, (cls.clone(), cls)),
+        }
     }
 }
 
 impl Constructor for PyClassMethod {
     type Args = PyObjectRef;
 
     fn py_new(cls: PyTypeRef, callable: Self::Args, vm: &VirtualMachine) -> PyResult {
-        PyClassMethod {
-            callable: PyMutex::new(callable),
+        let result: PyResult<PyObjectRef> = PyClassMethod {
+            callable: PyMutex::new(callable.clone()),
         }
         .into_ref_with_type(vm, cls)
```

Comment:
i'd like to propagate error here before later handling

```suggestion
        .into_ref_with_type(vm, cls)?
```